pkg/minishift/provisioner/minishift_detector_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright (C) 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioner
+
+import (
+	"testing"
+
+	"github.com/docker/machine/libmachine/provision"
+	"github.com/minishift/minishift/pkg/minikube/tests"
+	"github.com/docker/machine/libmachine/drivers"
+)
+
+func TestMinishiftDetector(t *testing.T){
+	s, _ := tests.NewSSHServer()
+	s.CommandToOutput = make(map[string]string)
+	s.CommandToOutput["cat /etc/os-release"] = `NAME="CentOS Linux"
+VERSION="7 (Core)"
+ID="centos"
+ID_LIKE="rhel fedora"
+VERSION_ID="7"
+PRETTY_NAME="CentOS Linux 7 (Core)"
+ANSI_COLOR="0;31"
+CPE_NAME="cpe:/o:centos:centos:7"
+HOME_URL="https://www.centos.org/"
+BUG_REPORT_URL="https://bugs.centos.org/"
+
+CENTOS_MANTISBT_PROJECT="CentOS-7"
+CENTOS_MANTISBT_PROJECT_VERSION="7"
+REDHAT_SUPPORT_PRODUCT="centos"
+REDHAT_SUPPORT_PRODUCT_VERSION="7"
+
+VARIANT="minishift"
+VARIANT_VERSION="1.0.0-alpha.1"
+`
+	port, err := s.Start()
+	if err != nil {
+		t.Fatalf("Error starting ssh server: %s", err)
+	}
+	d := &tests.MockDriver{
+		Port: port,
+		BaseDriver: drivers.BaseDriver{
+			IPAddress:  "127.0.0.1",
+			SSHKeyPath: "",
+		},
+	}
+
+	detector := MinishiftProvisionerDetector{Delegate: provision.StandardDetector{}}
+	provisioner, err := detector.DetectProvisioner(d)
+	if err != nil {
+		t.Fatalf("Error Getting detector: %s", err)
+	}
+	osRelease, _ := provisioner.GetOsReleaseInfo()
```

Comment:
I don't get it. You are not proofing anything here. _provisioner.GetOsReleaseInfo()_ is part of the interface. Any properly implemented provisioner will return the full release info and since as per test data you added the variant, this will always pass, regardless which actual type the provisioner has. What you need to check is the dynamic type of the provisioner:

    reflect.TypeOf(provisioner)

Once you get _*provisioner.MinishiftProvisioner_ and once _*provision.CentosProvisioner_. This is what you need to assert on. This tests and verifies the proper selection of a provisoner.